### PR TITLE
Add keybind for cycling grid type

### PR DESCRIPTION
**Pull request by OliBomby**
Requires:
- [x] https://github.com/ppy/osu/pull/26309
- [x] https://github.com/ppy/osu/pull/26310
- [x] https://github.com/ppy/osu/pull/26311
- [x] https://github.com/ppy/osu/pull/26313

Using the tool boxes on the right frequently is quite annoying, so I also added a keybind for cycling the grid type (H). I chose this key because its right next to the grid spacing cycle hotkey (G) and unused. It makes sense to group the grid property changing hotkeys physically close on the keyboard. An alternative hotkey could be like Shift+G or Shift+G, Ctrl+G is taken.

There's also a test for the grid placement tool in here which apparently got left behind when I was splitting up this PR into parts. I guess it can't hurt to have it so here it is.

https://github.com/ppy/osu/assets/17460441/cf63348d-0fdc-4170-b497-bae3c69cee7a

### CODE REVIEW

**Comment by bdach** [comment]
1.3k line diffstat moves this even lower on review priority list than it already was.

Did everything here *need* to be in a single pull?

**Comment by OliBomby** [comment]
Right I forgot to split it up to make review more palatable.

**Comment by bdach** [comment]
What's with the history here?

Please force-push that away because whatever that diffstat is, it's not worth 50 commits. If you chain PRs correctly the commits from previous PRs in the chain should go away when merging master into PRs later down the chain; and here it appears they didn't. So I'd rather this was fixed up, even if by squashing, because I don't want to see all that stuff in blame.

**Comment by OliBomby** [comment]
I'm struggling with git here to fix this history

**Comment by peppy** [comment]
I dunno what's going on with the new commits but this should be fixed by squashing down to a single commit.

**Comment by OliBomby** [comment]
Ok its fixed

**Comment by peppy** [comment]
I guess cycling is okay and saves on used hotkeys, but I'd bet the first thing we get asked for next is per-grid hotkeys rather than a cycle. Any thoughts on this?

**Comment by peppy** [comment]
I've pushed some changes and fixes. @OliBomby please confirm you're okay with them.

Importantly, you have to make sure to add new bindings to the end of the enum, else it offsets game bindings game-wide and breaks everything (see https://github.com/ppy/osu/pull/26303/commits/b62d9f8696206b9779c7d7286d974cd4e41fe581).

File: osu.Game.Rulesets.Osu.Tests/Editor/TestSceneOsuEditorGrids.cs [hunk 1]
```diff
@@ -189,5 +191,92 @@ private void nextGridSizeIs(int size)
...
                                                        && EditorBeatmap.BeatmapInfo.GridSize == size);
+
+        [Test]
+        public void TestGridTypeToggling()
+        {
+            AddStep("enable rectangular grid", () => InputManager.Key(Key.T));
+            AddUntilStep("rectangular grid visible", () => this.ChildrenOfType<RectangularPositionSnapGrid>().Any());
+            gridActive<RectangularPositionSnapGrid>(true);
+
+            nextGridTypeIs<TriangularPositionSnapGrid>();
```
[the remark this answers is not in the document]
Comment (OliBomby) [reply osu.Game.Rulesets.Osu.Tests/Editor/TestSceneOsuEditorGrids.cs:203]: I've added an assert to explicitly assert the specific grid type. Now it should fail on the assert if it has the wrong grid type.

**Review by bdach — APPROVED** [review]

**Review by peppy — APPROVED** [review]

**Pull request merged**